src/xscen/spatial.py

```diff
@@ -158,6 +161,73 @@ def _dot(arr, wei):
     )
 
 
+def rotate_vectors(
+    uu: xr.DataArray,
+    vv: xr.DataArray,
+    crs: cartopy.crs.Projection | None = None,
+    reverse: bool = False,
+) -> tuple[xr.DataArray, xr.DataArray]:
+    """
+    Rotate a vector field defined in its grid space to the real south-north / west-east axes.
+
+    Parameters
+    ----------
+    uu : xr.DataArray
+      The u component of the vector (along the X grid axis).
+      Must be CF-compliant, so its coordinate are correctly found.
+    vv : xr.DataArray
+      The v component of the vector (along the Y grid axis)
+      Must be CF-compliant, so its coordinate are correctly found.
+    crs : pyproj.CRS, optional
+      The projection of the UU and VV grid. If not given, the grid mapping attribute from UU is read.
+    reverse: bool
+      If True, the opposite rotation is performed : inputs are understood as in the real S-N, W-E axes and are
+      rotated back to the grid axes.
+
+    Returns
+    -------
+    uuc : xr.DataArray
+      The west-east component of the vector
+    vvc : xr.DataArray
+      The south-north component of the vector
+    """
+    if crs is None:
+        crs = get_crs(uu.rename("uu").to_dataset())
+
+    lalo = cartopy.crs.PlateCarree(globe=crs.globe)
+    geod = lalo.get_geod()
+
+    X = uu.cf["X"]
+    Y = uu.cf["Y"]
+    # ensure everything has the same dim order
+    YY, XX = xr.broadcast(Y, X)
+    XXarr = XX.values
+    YYarr = YY.values
+    # a very small increment along the y axis
+    if X.dims == Y.dims:
+        # we got a list of points, can't infer the grid resolution
+        dy = 1e-5
+    else:
+        dy = (Y[1] - Y[0]).item() / 1000
+
+    # Get lat lon of center points
+    crds_c = lalo.transform_points(crs, XXarr, YYarr)
+    # Get lat lon of points just slightly above along the y axis
+    crds_y = lalo.transform_points(crs, XXarr, YYarr + dy)
+    # The azimuth of this short vector
+    az, _, _ = geod.inv(crds_c[..., 0], crds_c[..., 1], crds_y[..., 0], crds_y[..., 1])
+    if reverse:
+        az = -az
+    # The rotation angle : opposite of the azimuth, in rad
+    th = xr.DataArray(-np.deg2rad(az), dims=YY.dims, coords=YY.coords)
+    # Rotation matrix like in Algèbre linéaire et géométrie vectoriel
```

Comment:
Should we have a real reference?
```suggestion
    # Rotation matrix like in Algèbre linéaire et géométrie vectorielle
```